logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java

```diff
@@ -56,19 +64,29 @@ public String getMethod() {
 
     @Override
     public String getScheme() {
-        // TODO pick the real one
-        return "http";
+        SslHandler handler = context.channel().pipeline().get(SslHandler.class);
+        return handler == null ? "http" : "https";
     }
 
     @Override
     public String getHost() {
-        return request.headers().get(HOST, "unknown");
+        String host = request.headers().get(HOST);
+        if (host != null) {
+            int separator = host.indexOf(":");
+            return separator == -1 ? host : host.substring(0, separator);
+        } else {
+            return extractAddress().map(NetUtil::getHostname).orElse("unknown");
```

Comment:
Is this safe to do? Does it to a DNS query? Would it actually report a hostname that the client didn't use?

Reply:
Yes, it is
This method delegates to `InetSocketAddress::getHostString` on java >= 1.7
Replaced with explicit usage of this method.

Reply:
I'm still thinking, under which circumstances would this code execute and what does the hostname actually represent in that case? It's essentially a heuristic for the case that we didn't receive a `Host` header, right? We then try to do a reverse lookup of the channel's ip address to find the most likely hostname?